tests/unit/einstein/test_ceph.py

```diff
@@ -0,0 +1,170 @@
+import time
+import unittest
+import subprocess
+from ims.common import config
+
+config.load()
+
+import rados
+import rbd
+from ims.common import constants
+from ims.einstein import ceph
+from ims.einstein.operations import BMI
+from ims.exception import file_system_exceptions
+from ims.common.log import trace
+import logging
+
+_cfg = config.get()
+
+POOL = _cfg.fs.pool
+# test image
+CEPH_IMG = "BMI_TEST_CEPH"
+# test image size
+CEPH_IMG_SIZE = 1024
+# test snap_image
+CEPH_EXISTING_IMG = "bmi_ci.img"
+CEPH_SNAP_IMG = "BMI_TEST_CEPH_SNAP"
+# test write
+TEST_DATA = 'bmi' * 10
+OFFSET = 0
+# test clone
+CEPH_CHILD_IMG = "BMI_TEST_CEPH_CHILD_IMAGE"
+
+
+class TestImage(unittest.TestCase):
+    @trace
+    def setUp(self):
+        self.fs = ceph.RBD(_cfg.fs, _cfg.iscsi.password)
+        self.fs.create_image(CEPH_IMG, CEPH_IMG_SIZE)
+        self.TEST_NOT_EXIST_IMG_NAME = "BMI_TEST_IMG_NOT_FOUND"
+
+    def TestGetImage(self):
+        """
+        Test get image
+        """
+        good_img = self.fs.get_image(CEPH_IMG)
+        self.assertEqual(good_img.name, CEPH_IMG)
+        with self.assertRaises(file_system_exceptions.ImageNotFoundException):
+            bad_img = self.fs.get_image(self.TEST_NOT_EXIST_IMG_NAME)
+
+    def TestListImages(self):
+        """
+        Test create image and list images
+        """
+        imgs = self.fs.list_images()
+        self.assertIn(CEPH_IMG, imgs)
+        command = "echo {0} | sudo -S rbd --keyring {1} --id {2} ls " \
+                  "{3}".format(self.fs.password, self.fs.keyring,
+                               self.fs.rid, self.fs.pool)
+        p = subprocess.Popen(command, shell=True, stderr=subprocess.STDOUT,
+                             stdout=subprocess.PIPE)
+        output, err = p.communicate()
+        rbd_imgs = output.split()
+        self.assertListEqual(imgs, rbd_imgs)
+
+    def TestWrite(self):
+        """
+        Test write image
+        """
+        bytes_written = self.fs.write(CEPH_IMG, TEST_DATA, OFFSET)
+        data_read = self.fs.read(CEPH_IMG, OFFSET, bytes_written)
+        self.assertEqual(TEST_DATA, data_read)
+
+    def TestRemoveImage(self):
+        """
+        Test remove image and remove snapshot
+        """
+        isRemoved = self.fs.remove(CEPH_IMG)
+        self.assertTrue(isRemoved)
+        imgs = self.fs.list_images()
+        self.assertNotIn(CEPH_IMG, imgs)
+
+    def runTest(self):
+        self.TestGetImage()
+        self.TestListImages()
+        self.TestWrite()
```

Comment:
You don't need to call all the functions here. Remove this function. The test framework will call each your functions